test/jdk/sun/net/www/http/KeepAliveCache/B8291637.java

```diff
@@ -125,28 +127,35 @@ public void run() {
         }
     }
 
-    public static void main(String[] args) throws Exception {
-        Server server = new Server(args[0]);
-        int port = server.getPort();
-        server.start();
-        URL url = new URL("http://127.0.0.1:" + Integer.toString(port) + "/");
-        HttpURLConnection urlc = (HttpURLConnection) url.openConnection();
-        InputStream i = urlc.getInputStream();
-        int c,count=0;
-        byte[] buf = new byte[256];
-        while ((c=i.read(buf)) != -1) {
-            count+=c;
-        }
-        i.close();
-        System.out.println("Read " + count );
-        try {
+    public static void runTest(String param) throws Exception {
+        try (Server server = new Server(param)) {
+            server.start();
+            URL url = URIBuilder.newBuilder()
+                    .scheme("http")
+                    .loopback()
+                    .port(server.getPort())
+                    .path("/")
+                    .toURL();
+            HttpURLConnection urlc = (HttpURLConnection) url.openConnection();
+            int count = 0;
+            try (InputStream i = urlc.getInputStream()) {
+                int c;
+                byte[] buf = new byte[256];
+                while ((c = i.read(buf)) != -1) {
```

Comment:
Given that the server response is just a hello world message, I think you can just replace this entire code within the try block with something like:

```java
int count;
try (InputStream i = urlc.getInputStream()) {
    count = i.readAllBytes().length;
}
```

If you prefer to leave it in the current form, that's fine too.

Reply:
Done.